src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Desktop.cs

```diff
@@ -120,27 +121,25 @@ internal bool EnsureResolvedUnhooked()
                 const string resourcesExtension = ".resources";
                 var assemblyName = new AssemblyName(args.Name);
                 var simpleName = assemblyName.Name;
-                var isSatelliteAssembly =
-                    assemblyName.CultureInfo is not null &&
-                    simpleName.EndsWith(resourcesExtension, StringComparison.Ordinal);
 
-                if (isSatelliteAssembly)
+                string? loadPath;
+                if (assemblyName.CultureInfo is not null && simpleName.EndsWith(resourcesExtension, StringComparison.Ordinal))
                 {
                     // Satellite assemblies should get the best path information using the
                     // non-resource part of the assembly name. Once the path information is obtained
```

Comment:
> Satellite assemblies should get the best path information using the non-resource part of the assembly name.

Given this statement, should we assert inside GetBestResolvedPath that `assemblyName.CultureInfo is null`? Also looks like GetBestResolvedPath might be called from GetOriginalDependencyLocation for satellite assemblies as well, shouldn't that be avoided?

Reply:
Consider the case when a satellite assembly was directly passed with `/analyzer`. In that case both of these methods would need to function with satellite assembly names. I'm honestly not sure if we should care about this or not.